drivers/usb/udc/udc_common.c

```diff
@@ -46,6 +46,31 @@ struct udc_ep_config *udc_get_ep_cfg(const struct device *dev, const uint8_t ep)
 	return data->ep_lut[USB_EP_LUT_IDX(ep)];
 }
 
+bool udc_ep_is_busy(const struct device *dev, const uint8_t ep)
+{
+	struct udc_ep_config *ep_cfg;
+
+	ep_cfg = udc_get_ep_cfg(dev, ep);
+	if (unlikely(ep_cfg == NULL)) {
+		__ASSERT(false, "ep 0x%02x is not available", ep);
+		return true;
+	}
+
+	return ep_cfg->stat.busy;
+}
+
+void udc_ep_set_busy(const struct device *dev, const uint8_t ep, const bool busy)
+{
+	struct udc_ep_config *ep_cfg;
+
+	ep_cfg = udc_get_ep_cfg(dev, ep);
+	if (unlikely(ep_cfg == NULL)) {
+		__ASSERT(false, "ep 0x%02x is not available", ep);
+	} else {
+		ep_cfg->stat.busy = busy;
+	}
+}
```

Comment:
I would definitely prefer `__ASSERT(ep_cfg != NULL, "ep 0x%02x is not available", ep);` followed by unconditional `ep_cfg->stat.busy = busy;`. This way if asserts are disabled the code will simply fail writing to NULL pointer and will make the issue obvious (which will save a lot of time debugging it later).